include/core/thermal_conductivity_model.h

```diff
@@ -188,4 +188,122 @@ class ThermalConductivityLinear : public ThermalConductivityModel
+   * @return value of the thermal conductivity calculated with the fields_value.
+   */
+  double
+  value(const std::map<field, double> &fields_value) override
+  {
+    double thermal_conductivity;
+    // Thermal conductivity of solid phase
+    if (fields_value.at(field::temperature) < T_solidus)
+      thermal_conductivity = thermal_conductivity_s;
+    // Thermal conductivity of liquid phase
+    else if (fields_value.at(field::temperature) > T_liquidus)
+      thermal_conductivity = thermal_conductivity_l;
+    // Mean value of the conductivities of the solid and liquid phases
+    else
+      {
+        const double l_frac =
+          std::min(std::max((fields_value.at(field::temperature) - T_solidus) /
+                              (T_liquidus - T_solidus),
+                            0.),
+                   1.);
+
+        thermal_conductivity = thermal_conductivity_l * l_frac +
+                               thermal_conductivity_s * (1. - l_frac);
+      }
+
+    return thermal_conductivity;
+  };
+
+  /**
+   * @brief vector_value Calculates the vector value of thermal conductivities
+   * @param field_vectors Vector of properties on which the thermal conductivities depend
+   * @param property_vector Values of the thermal conductivities
+   */
+  void
+  vector_value(const std::map<field, std::vector<double>> &field_vectors,
+               std::vector<double> &property_vector) override
+  {
+    const std::vector<double> &T = field_vectors.at(field::temperature);
+    for (unsigned int i = 0; i < property_vector.size(); ++i)
+      {
+        // Thermal conductivity of solid phase
+        if (T[i] < T_solidus)
+          property_vector[i] = thermal_conductivity_s;
+        // Thermal conductivity of liquid phase
+        else if (T[i] > T_liquidus)
+          property_vector[i] = thermal_conductivity_l;
+        else
+          {
+            const double l_frac = std::min(
+              std::max((T[i] - T_solidus) / (T_liquidus - T_solidus), 0.), 1.);
+
+            property_vector[i] = thermal_conductivity_l * l_frac +
+                                 thermal_conductivity_s * (1. - l_frac);
+          }
+      }
+  };
+
+  /**
+   * @brief jacobian Calcualtes the jacobian (the partial derivative) of the thermal conductivity with respect to a field
+   * @param field_values Value of the various fields on which the property may depend.
+   * @param id Indicator of the field with respect to which the jacobian
+   * should be calculated
+   * @return value of the partial derivative of the thermal conductivity with respect to the field.
+   */
+
+  double
+  jacobian(const std::map<field, double> &field_values, field id) override
+  {
+    if (id == field::temperature)
+      return this->numerical_jacobian(field_values, field::temperature);
+    else
+      return 0;
+  };
+
+  /**
+   * @brief vector_jacobian Calculate the derivative of the with respect to a field
```

Comment:
```suggestion
   * @brief vector_jacobian Calculate the derivative of the ? (viscosity?) with respect to a field
```

Reply:
Should be thermal conductivity here not viscosity